packages/forms-web-app/tests/unit/controllers/eligibility/listed-building.test.js

```diff
@@ -19,54 +35,100 @@ describe('controllers/eligibility/listed-building', () => {
       listedBuildingController.getListedBuilding(req, res);
 
       expect(res.render).toHaveBeenCalledWith(VIEW.ELIGIBILITY.LISTED_BUILDING, {
+        appeal: req.session.appeal,
         FORM_FIELD: listedBuildingController.FORM_FIELD,
       });
     });
   });
 
   describe('postListedBuilding', () => {
-    it('should redirect on the happy path', () => {
+    it('should re-render the template with errors if there is any validation error', async () => {
       const mockRequest = {
-        body: {},
+        ...req,
+        body: {
+          'is-your-appeal-about-a-listed-building': 'bad value',
+          errors: { a: 'b' },
+          errorSummary: [{ text: 'There were errors here', href: '#' }],
+        },
       };
-      const mockResponse = {
-        redirect: jest.fn(),
+      await listedBuildingController.postListedBuilding(mockRequest, res);
+
+      expect(createOrUpdateAppeal).not.toHaveBeenCalled();
```

Comment:
This is not testing re-rendering, should this be in it's own explicitly named test?

Reply:
These steps (this and the other two comments that relate) are testing the function calls that happen before the re-rendering. If I split them out I would have something like this:

(assume we have five steps in the controller)

test 1:

1. controller call
2. sub func 1 is called

test 2:

1. controller call
2. sub func 1 is called
3. sub func 2 is called

test 2:

1. controller call
2. sub func 1 is called
3. sub func 2 is called
4. sub func 3 is called

test 2:

1. controller call
2. sub func 1 is called
3. sub func 2 is called
4. sub func 3 is called
5. sub func 4 is called

test 2:

1. controller call
2. sub func 1 is called
3. sub func 2 is called
4. sub func 3 is called
5. sub func 4 is called
6. sub func 5 is called

Seems like a lot of repeated tests to get to the same outcome. 

But equally, I feel like I have misunderstood?